confluent_kafka/avro/serializer/message_serializer.py

```diff
@@ -165,19 +167,24 @@ def _get_decoder_func(self, schema_id, payload):
             raise SerializerError("unable to fetch schema with id %d" % (schema_id))
 
         curr_pos = payload.tell()
+
+        reader_schema_obj = self.reader_key_schema if is_key else self.reader_value_schema
+
         if HAS_FAST:
             # try to use fast avro
             try:
-                schema_dict = schema.to_json()
-                schemaless_reader(payload, schema_dict)
+                writer_schema = schema.to_json()
```

Comment:
To keep naming consistent we should replace schema(declared a few lines up on 162) with `writer_schema_obj`

Reply:
Actually kept it intentionally as `schema` to be consistent with method param `schema_id`, but you are right, changed it: 70453e8
(kept `schema_id` as is though since there is no reader `schema_id` option; could be added later as alternative)